version 4/03 environments/04 Production Configuration/README.md

```diff
@@ -0,0 +1,341 @@
+
+- Let's go to rename `webpack.config.js` to `base.webpack.config.js`:
+
+### ./base.webpack.config.js
+```diff
+var HtmlWebpackPlugin = require('html-webpack-plugin');
+var MiniCssExtractPlugin = require('mini-css-extract-plugin');
+- var webpack = require('webpack');
+var path = require('path');
+
+var basePath = __dirname;
+
+module.exports = {
+  context: path.join(basePath, 'src'),
+  resolve: {
+    extensions: ['.js', '.jsx', '.scss'],
+  },
+  entry: {
+    app: './students.jsx',
+    vendor: [
+      'react',
+      'react-dom',
+    ],
+    vendorStyles: [
+      '../node_modules/bootstrap/dist/css/bootstrap.css',
+    ],
+  },
+  output: {
+    filename: '[name].[hash].js',
+  },
+  optimization: {
+    splitChunks: {
+      cacheGroups: {
+        vendor: {
+          chunks: 'initial',
+          name: 'vendor',
+          test: 'vendor',
+          enforce: true
+        },
+      }
+    }
+  },
+  module: {
+    rules: [
+      {
+        test: /\.jsx$/,
+        exclude: /node_modules/,
+        loader: 'babel-loader',
+      },
+      {
+        test: /\.scss$/,
+        use: [
+          MiniCssExtractPlugin.loader,
+          { 
+            loader: "css-loader",
+            options: {
+              modules: true,
+              localIdentName: '[name]__[local]___[hash:base64:5]',
+              camelCase: true,
+            },    
+          },
+          "sass-loader",
+        ]
+      },
+      {
+        test: /\.css$/,
+        use: [
+          MiniCssExtractPlugin.loader,
+          "css-loader"
+        ]
+      },      
+    ],
+  },
+  plugins: [
+    //Generate index.html in /dist => https://github.com/ampedandwired/html-webpack-plugin
+    new HtmlWebpackPlugin({
+      filename: 'index.html', //Name of file in ./dist/
+      template: 'index.html', //Name of template in ./src
+      hash: true,
+    }),
+    new MiniCssExtractPlugin({
+      filename: "[name].css",
+      chunkFilename: "[id].css"
+    }),
+  ],
+};
+```
+
+- Now it's time to install `webpack-merge` package. This allow us to combine `base.webpack.config` with environment config:
+
+```
+npm install webpack-merge --save-dev
+```
+
+- Let's go to create the `dev` environment configuration:
+
+### ./dev.webpack.config.js
+```javascript
+const merge = require('webpack-merge');
+const common = require('./base.webpack.config.js');
+
+module.exports = merge(common, {
+    mode: 'development',
+    devtool: 'inline-source-map',
+    devServer: {
+        contentBase: './dist'
+    }
+});
+```
+
+- Make a configuration to see in console the enviroment in which we are
```

Comment:
Just to make a quick check we will add console log, to check in which environment we are running the app (dev or production)

Reply:
done